src/Raven.Server/Documents/ETL/Providers/Elasticsearch/ElasticsearchEtl.cs

```diff
@@ -0,0 +1,182 @@
+        {
+            throw new NotSupportedException("Counters aren't supported by ElasticSearch ETL");
+        }
+
+        protected override IEnumerator<ElasticSearchItem> ConvertTimeSeriesEnumerator(DocumentsOperationContext context, IEnumerator<TimeSeriesSegmentEntry> timeSeries,
+            string collection)
+        {
+            throw new NotSupportedException("Time series aren't supported by ElasticSearch ETL");
+        }
+
+        protected override IEnumerator<ElasticSearchItem> ConvertTimeSeriesDeletedRangeEnumerator(DocumentsOperationContext context,
+            IEnumerator<TimeSeriesDeletedRangeItem> timeSeries, string collection)
+        {
+            throw new NotSupportedException("Time series aren't supported by ElasticSearch ETL");
+        }
+
+        protected override EtlTransformer<ElasticSearchItem, ElasticSearchIndexWithRecords, EtlStatsScope, EtlPerformanceOperation> GetTransformer(DocumentsOperationContext context)
+        {
+            return new ElasticSearchDocumentTransformer(Transformation, Database, context, Configuration);
+        }
+
+        protected override int LoadInternal(IEnumerable<ElasticSearchIndexWithRecords> records, DocumentsOperationContext context, EtlStatsScope scope)
+        {
+            int statsCounter = 0;
+
+            if (_client == null)
+            {
+                _client = ElasticSearchHelper.CreateClient(Configuration.Connection);    
+            }
+
+            StringBuilder deleteQuery = new StringBuilder();
+            
+            foreach (var index in records)
+            {
+                deleteQuery.Clear();
+                
+                foreach (ElasticSearchItem delete in index.Deletes)
+                {
+                    deleteQuery.Append($"{delete.DocumentId},");
+                }
+
+                var deleteResponse = _client.DeleteByQuery<string>(d => d
+                    .Index(index.IndexName.ToLower())
+                    .Query(q => q
+                        .Match(p => p
+                            .Field(index.IndexIdProperty)
+                            .Query(deleteQuery.ToString()))
+                    )
+                );
+
+                if (string.IsNullOrWhiteSpace(deleteResponse.DebugInformation) == false)
+                {
+                    throw new ElasticSearchLoadException($"Index {index}; Documents IDs: {deleteQuery}; error: {deleteResponse.DebugInformation}");
+                }
+                
+                if (deleteResponse.ServerError != null)
+                {
+                    throw new ElasticSearchLoadException($"Index {index}; Documents IDs: {deleteQuery}; error: {deleteResponse.ServerError.Error}");
+                }
+                
+                if (deleteResponse.OriginalException != null)
+                {
+                    throw new ElasticSearchLoadException($"Index {index}; Documents IDs: {deleteQuery}; error: {deleteResponse.OriginalException}");
```

Comment:
Maybe move the exception throwing to a dedicated method and just pass `index` and `DebugInformation / ServerError.Error / OriginalException`